### ISSUE TITLE
Allow weaving of generic Interfaces

### ISSUE THREAD

**Issue by BrunoJuchli** [BrunoJuchli/StaticProxy.Fody#6]
Imagine an interface like

```
IFactory<T>
    : where T : Foo
{
    T Create(string parameter);
}
```

StaticProxy.Fody should create an open-generic implementation. The consumer (integration package...) can then decide which closed types to bind and how.

---
- How can the interceptor access the generic type arguments?
---> concrete object available to StaticProxy.Intercepter by `implementationTarget`
------> but made available to the interceptor how?

--> Also see preliminary unit tests @ When_proxying_interface_with_generic_parameters

**Comment by BrunoJuchli** [comment]
Duplicated by #12

**Comment by BrunoJuchli** [comment]
Implemented by #14
